e2e/wdio/core/tests/dialog.e2e-spec.ts

```diff
@@ -0,0 +1,676 @@
+    dismissedStatus, dottedOutline,
+    escapeStatus, fullscreenClass, mobileClass,
+    mobileProperty, noMobileSpacingClass, noOutline, outlineProperty,
+    styleAttribute, topPaddingProperty
+} from '../fixtures/appData/dialog-contents';
+import { papayaFruit } from '../fixtures/testData/dialog';
+
+describe('dialog test suite', function() {
+    const dialogPage = new DialogPo();
+    const {
+        templateDialog, button, dialog, dialogOutput, componentDialog, objectDialog, stateDialog, busyIndicator,
+        configurationDialog, dialogContainer, resizeHandle, positionDialog, mobileDialog, complexDialog, dialogItems,
+        searchBar, dialogCartOutput, stackedDialog, playgroundDialog, checkboxes, inputFields, dialogExamples,
+        customDialog, dialogBody
+    } = dialogPage;
+
+    beforeAll(() => {
+        dialogPage.open();
+    }, 1);
+
+    describe('template based dialog example', function() {
+        it('should check dialog dismissal and output', () => {
+            const closeBtn = 0;
+            const acceptBtn = 1;
+            const cancelBtn = 2;
+
+            checkDialogDismissals(templateDialog, button, closeBtn, closeBtnStatus);
+            checkDialogDismissals(templateDialog, button, acceptBtn, continueStatus);
+            checkDialogDismissals(templateDialog, button, cancelBtn, canceledStatus);
+            checkCloseDialogWithEscapeKey(templateDialog, button);
+        });
+    });
+
+    describe('component based dialog example', function() {
+        it('should check dialog dismissal and output', () => {
+            const closeBtn = 0;
+            const acceptBtn = 1;
+            const cancelBtn = 2;
+
+            checkDialogDismissals(componentDialog, button, closeBtn, closeBtnStatus);
+            checkDialogDismissals(componentDialog, button, acceptBtn, continueStatus);
+            checkDialogDismissals(componentDialog, button, cancelBtn, canceledStatus);
+            checkCloseDialogWithEscapeKey(componentDialog, button);
+        });
+    });
+
+    describe('object based dialog example', function() {
+        it('should check dialog dismissal and output', () => {
+            const closeBtn = 0;
+            const acceptBtn = 2;
+            const cancelBtn = 3;
+
+            checkDialogDismissals(objectDialog, button, closeBtn, dismissedStatus);
+            checkDialogDismissals(objectDialog, button, acceptBtn, approvedStatus);
+            // skipped due to issue: https://github.com/SAP/fundamental-ngx/issues/4967
+            // checkDialogDismissals(objectDialog, button, cancelBtn, canceledStatus);
+            checkCloseDialogWithEscapeKey(objectDialog, button);
+        });
+    });
+
+    describe('dialog state examples', function() {
+        it('should check auto dismissal', () => {
+            const selfDismissingDialogCount = 3;
+
+            for (let i = 0; i < selfDismissingDialogCount; i++) {
+                openDialog(stateDialog, i);
+                pause(4500);
```

Comment:
4,5s  ? Maybe it is a better way to do it? 
If dialog is not appeared after 2-3 s please report the bug.

Reply:
'open dialog' function has waitForElDisplayed in it.
the pause is to make sure that dialog closes automatically. It should close in 4 seconds. 
So this function opens the dialog, makes sure it's displayed, waits for animation (dialog to close automatically) and checks dialog is closed.

Reply:
Maybe  waitForNotPresent then ?